end_to_end/end_to_end_suite_test.go

```diff
@@ -841,6 +841,43 @@ var _ = Describe("backup and restore end to end tests", func() {
 			}
 			Expect(err).NotTo(HaveOccurred())
 		})
+		It(`Can backup and concurrent restores of same backup with helpers`, func() {
+			testhelper.AssertQueryRuns(backupConn,
+				"CREATE TABLE public.t (i int) DISTRIBUTED BY (i)")
+			defer testhelper.AssertQueryRuns(backupConn, "DROP TABLE public.t")
+			output := gpbackup(gpbackupPath, backupHelperPath,
+				"--single-data-file", "--include-table", "public.t")
+			timestamp := getBackupTimestamp(string(output))
+			gprestore(gprestorePath, restoreHelperPath, timestamp,
+				"--redirect-db", "restoredb", "--metadata-only")
+			restoreConn.MustExec(
+				"BEGIN; LOCK TABLE public.t IN ACCESS EXCLUSIVE MODE")
+			var backWg sync.WaitGroup
+			errchan := make(chan error, 2)
+			backWg.Add(1)
+			go func() {
+				defer backWg.Done()
+				cmd := exec.Command(gprestorePath, "--timestamp", timestamp,
+					"--redirect-db", "restoredb", "--data-only")
+				_, err := cmd.CombinedOutput()
+				errchan <- err
+			}()
+			backWg.Add(1)
+			go func() {
+				defer backWg.Done()
+				cmd := exec.Command(gprestorePath, "--timestamp", timestamp,
+					"--redirect-db", "restoredb", "--data-only",
+					"--exclude-table", "public.t")
+				_, err := cmd.CombinedOutput()
+				restoreConn.MustExec("COMMIT")
+				errchan <- err
+			}()
+			backWg.Wait()
```

Comment:
Maybe check that public.t inside restoredb contain correct data after recovery?

Reply:
that table is empty

Reply:
Maybe use no  empty table to ensure that recovery was correct?

Reply:
added